#!/bin/bash

#If anything fails, we back out
set -e
if [ -z $BB_SCRIPT_DIR ]; then
  echo "[ERROR]: buildahbash script running outside buildahbash environment"
  exit 1
fi
source $BB_SCRIPT_DIR/buildah.funcs
COMPILE_STAGE_NAME="${BB_PKG_IMAGE_FQNAME}_build"
NAME="${BB_PKG_NAME}"

if [ -z ${N_CORE} ]; then
    N_CORE=4
fi

if ! podman image exists ${COMPILE_STAGE_NAME}; then

  BUILD=$(bb_from_buildbox)

  bb_cont_mkdir_cd ${BUILD} /opt/${NAME}-src

  buildah run ${BUILD} wget --no-check-certificate \
          http://root.cern.ch/download/pythia6.tar.gz
  buildah run ${BUILD} wget --no-check-certificate \
           http://www.hepforge.org/archive/pythia6/pythia-6.4.28.f.gz

  buildah run ${BUILD} gunzip pythia-6.4.28.f.gz
  buildah run ${BUILD} tar xfvz pythia6.tar.gz
  buildah run ${BUILD} mv pythia-6.4.28.f pythia6/pythia6428.f
  buildah run ${BUILD} rm pythia6/pythia6416.f
  bb_cont_cd ${BUILD} /opt/${NAME}-src/pythia6


  # Fix ODR violation that only seems to turn up in recent gcc.
  buildah run ${BUILD}  sed -i "s/int /extern int /g" pythia6_common_address.c
  buildah run ${BUILD}  sed -i "s/char /extern char /g" pythia6_common_address.c

  # but put this one back as it doesn't actually exist...
  buildah run ${BUILD}  sed -i "s/extern int pyuppr/int pyuppr/g" pythia6_common_address.c

  buildah run ${BUILD} bash makePythia6.linuxx8664

  bb_cont_mkdir ${BUILD} /opt/${NAME}/${BB_PKG_VERS}
  buildah run ${BUILD} mv libPythia6.so \
      /opt/${NAME}/${BB_PKG_VERS}/libPythia6.so.${BB_PKG_VERS}

  buildah run ${BUILD} ln -s \
    /opt/${NAME}/${BB_PKG_VERS}/libPythia6.so.${BB_PKG_VERS} \
    /opt/${NAME}/${BB_PKG_VERS}/libPythia6.so

  bb_commit ${BUILD} ${COMPILE_STAGE_NAME}

fi

BASE=$(bb_from_base ${BB_PKG_NAME} ${BB_PKG_VERS} $(bb_get_runbox ${BB_PKG_ROOT_IT}))

bb_consume_target ${NAME} ${BB_PKG_VERS} \
  $(buildah from ${COMPILE_STAGE_NAME}) ${BASE}

bb_commit ${BASE}
